File: Java/jobspls/algorithms/PermutationCheck.java
package jobspls.algorithms;

import java.util.Arrays;
import java.util.TreeSet;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class PermutationCheck {
    public static boolean perm_check(int[] haystack) {
        var numbers = new TreeSet<Integer>(Arrays
            .stream(haystack)
            .boxed()
            .collect(Collectors.toList()));

        return IntStream.range(1, haystack.length + 1)
                        .allMatch(n -> { return numbers.contains(n); });
    }
}

File: Go/jobspls/algorithms/permutation_check.go
package algorithms

func PermCheck(haystack []int) bool {
	numbers := make(map[int]struct{})
	for _, v := range haystack {
		numbers[v] = struct{}{}
	}

	for i := 1; i < len(haystack)+1; i++ {
		if _, found := numbers[i]; !found {
			return false
		}
	}

	return true
}
